### DISCUSSION TITLE
Can I get the value outside of promise?

### DISCUSSION THREAD

**Discussion by darrenli912** [mojolicious/mojo#1826] [Q&A] [answered]
I am refactoring my mojolicious app from block to unblock for some method.
For example:

sub a {
    my $a = block_function_a;
    my $b = block_function_b;
    my $c = block_function_c;
   return $a+$b+$c;
}

to 
sub a {
    my $a = block_function_a;
    my $b = block_function_b;
    return promise->then(sub($c){
    $a+$b+$c;
  } );
}
After the change , I  return value from $a+$b+$c to promise. It is annoying, because I have to change many relative code. 
Is there any way to return $a+$b+$c, even though I use some promise?

PS. promise->wait can't resolve the problem in my mojolicious app.

**Comment by kraih** [comment, marked as answer]
No, that is impossible.

**Comment by jberger** [comment]
The closest way to accomplish it with the least (but not no) code changes is to use async/await

**Reply by kraih** [reply]
That still requires the code to be surrounded by a promise that is handled without async/await.

**Reply by jberger** [reply]
Very true. I was responding mainly to the final lines:

> It is annoying, because I have to change many relative code.
> Is there any way to return $a+$b+$c, even though I use some promise?
> 
> PS. promise->wait can't resolve the problem in my mojolicious app.

Wherein the OP was mentioning how much change would be needed for adding promise handling everywhere, and the fact that they were already in a non-blocking mojo app (thus ->wait won't help). In this case, adding async/await is not no code but should be able to be gradually included more easily than in other places. (I know you know all this kraih, just mentioning for the future reader).
